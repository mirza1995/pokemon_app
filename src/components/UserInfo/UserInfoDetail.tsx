import { FormattedMessage } from 'react-intl';

type UserInfoDetailProps = {
  labelId: string;
  value: string;
};

export const UserInfoDetail = ({ labelId, value }: UserInfoDetailProps) => {
  return (
    <div className='mt-7'>
      <div className='text-xs font-normal text-gray'>
        <FormattedMessage id={labelId} />
      </div>
      <h5 className='mt-4 text-title font-normal'>{value}</h5>
    </div>
  );
};
