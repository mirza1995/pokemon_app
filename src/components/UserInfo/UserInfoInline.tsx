import { CurrentUser } from '@models/CurrentUser';
import userImage from '@images/user.jpg';

type UserInfoInlineProps = {
  user: CurrentUser;
  onUserClick: () => void;
};

export const UserInfoInline = ({ user, onUserClick }: UserInfoInlineProps) => {
  return (
    <div className='flex justify-start font-medium max-md:flex-col'>
      <button onClick={onUserClick} className='nav-link text-left md:mr-5'>
        {`${user.firstName} ${user.lastName}`}
      </button>

      <img className='max-md:mt-4' src={userImage} width='40' />
    </div>
  );
};
