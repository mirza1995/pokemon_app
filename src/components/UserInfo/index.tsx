export * from './UserInfo';
export * from './UserInfoDetail';
export * from './UserInfoInline';
