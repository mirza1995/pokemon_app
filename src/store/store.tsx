import {
  combineReducers,
  configureStore,
  createSelector
} from '@reduxjs/toolkit';
import userSlice from './user-slice';

const rootReducer = combineReducers({ user: userSlice.reducer });
const store = configureStore({
  reducer: rootReducer
});
export type StoreRootState = ReturnType<typeof rootReducer>;

const selectSelf = (state: StoreRootState) => state;

export const userSelector = createSelector(selectSelf, (state) => state.user);

export default store;
