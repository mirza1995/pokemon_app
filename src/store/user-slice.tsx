import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { UserSlice } from '@models/UserSlice';

const initialState: UserSlice = {
  user: null,
  authToken: ''
};

const userSlice = createSlice({
  name: 'user',
  initialState: initialState,
  reducers: {
    init(state: UserSlice, action: PayloadAction<UserSlice>) {
      state.user = action.payload.user;
      state.authToken = action.payload.authToken;
    },
    reset(state: UserSlice) {
      state.user = initialState.user;
      state.authToken = initialState.authToken;
    }
  }
});

export const userActions = userSlice.actions;

export default userSlice;
