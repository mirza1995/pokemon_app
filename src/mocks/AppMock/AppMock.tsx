import { Provider } from 'react-redux';
import { CookiesProvider } from 'react-cookie';
import { IntlProvider } from 'react-intl';
import English from 'lang/en.json';
import store from 'store/store';

//Mocks application to be used in tests
export const AppMock = ({ children }: { children: React.ReactNode }) => {
  return (
    <Provider store={store}>
      <CookiesProvider>
        <IntlProvider locale={navigator.language} messages={English}>
          {children}
        </IntlProvider>
      </CookiesProvider>
    </Provider>
  );
};
