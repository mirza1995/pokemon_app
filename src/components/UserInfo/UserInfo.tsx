import { useDispatch, useSelector } from 'react-redux';
import { useCookies } from 'react-cookie';
import { FormattedMessage } from 'react-intl';

import { userSelector } from '@store/store';
import { userActions } from '@store/user-slice';

import { PrimaryButton } from '@components/PrimaryButton';
import userImage from '@images/user.jpg';
import { UserInfoDetail } from './UserInfoDetail';
import { AUTH_TOKEN } from '@utils/cookies';

type UserInfoProps = {
  closeModal: () => void;
};

export const UserInfo = ({ closeModal }: UserInfoProps) => {
  const { user } = useSelector(userSelector);
  const dispatch = useDispatch();
  const removeCookie = useCookies([AUTH_TOKEN])[2];

  const onLogoutClick = () => {
    removeCookie(AUTH_TOKEN);
    dispatch(userActions.reset());
    closeModal();
  };

  return (
    <div className='modal-default'>
      {user ? (
        <div className='px-[18px] pb-[30px] md:px-[100px] md:pt-[40px]'>
          <div className='flex items-center'>
            <img src={userImage} width='80' />

            <div className='ml-7'>
              <h2 className='text-xl font-light'>
                {`${user.firstName} ${user.lastName}`}
              </h2>
              <p className='mt-[6px] text-md font-normal text-gray'>
                {/* TODO - update with real sightings number */}
                <FormattedMessage
                  id='sightings.plural'
                  values={{ number: 0 }}
                />
              </p>
            </div>
          </div>

          <div className='pt-2.5'>
            <UserInfoDetail
              labelId='profile.firstName'
              value={user.firstName}
            />
            <UserInfoDetail labelId='profile.lastName' value={user.lastName} />
          </div>

          <div className='mx-auto w-[147px] pt-[80px] md:pt-[70px]'>
            <PrimaryButton onClick={onLogoutClick} textId='profile.logout' />
          </div>
        </div>
      ) : (
        <h2>
          <FormattedMessage id='profile.notLoggedIn' />
        </h2>
      )}
    </div>
  );
};
