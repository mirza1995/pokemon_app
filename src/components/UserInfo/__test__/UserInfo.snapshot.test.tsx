/* eslint-disable @typescript-eslint/no-empty-function */
import renderer from 'react-test-renderer';
import { AppMock } from '@mocks/AppMock';
import store from '@store/store';
import { userActions } from '@store/user-slice';
import { UserInfo } from '@components/UserInfo';

it('User Info component renders correctly', async () => {
  await store.dispatch(
    userActions.init({
      user: {
        id: 1,
        firstName: 'Name',
        lastName: 'Surname'
      },
      authToken: 'authToken'
    })
  );

  const tree = renderer
    .create(
      <AppMock>
        <UserInfo closeModal={() => {}} />
      </AppMock>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
