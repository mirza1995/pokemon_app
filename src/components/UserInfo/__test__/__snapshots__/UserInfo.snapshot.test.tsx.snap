// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Info component renders correctly 1`] = `
<div
  className="modal-default"
>
  <div
    className="px-[18px] pb-[30px] md:px-[100px] md:pt-[40px]"
  >
    <div
      className="flex items-center"
    >
      <img
        src=""
        width="80"
      />
      <div
        className="ml-7"
      >
        <h2
          className="text-xl font-light"
        >
          Name Surname
        </h2>
        <p
          className="mt-[6px] text-md font-normal text-gray"
        >
          0 sightings
        </p>
      </div>
    </div>
    <div
      className="pt-2.5"
    >
      <div
        className="mt-7"
      >
        <div
          className="text-xs font-normal text-gray"
        >
          First Name
        </div>
        <h5
          className="mt-4 text-title font-normal"
        >
          Name
        </h5>
      </div>
      <div
        className="mt-7"
      >
        <div
          className="text-xs font-normal text-gray"
        >
          Last Name
        </div>
        <h5
          className="mt-4 text-title font-normal"
        >
          Surname
        </h5>
      </div>
    </div>
    <div
      className="mx-auto w-[147px] pt-[80px] md:pt-[70px]"
    >
      <button
        className="w-full rounded-3 bg-primary-light py-4.5 text-base text-white shadow-3xl"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>
</div>
`;
